src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -148,10 +148,10 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
 
         public static void AddCommonServices(IServiceCollection services)
         {
+            services.AddSingleton<IHostIdProvider, ScriptHostIdProvider>();
```

Comment:
This needs to use `AddSingleton` as we have a default provider registered by the SDK.